// locationHandler/index.js

const { LocationClient, SearchPlaceIndexForTextCommand, CalculateRouteCommand } = require("@aws-sdk/client-location");
const locationClient = new LocationClient();

module.exports.getPlaceFromAddress = async (address, placeIndex) => {
    const params = {
        IndexName: placeIndex,
        Text: address
    }
    const response = await locationClient.send(new SearchPlaceIndexForTextCommand(params));
    console.log(response);
    return response.Results[0].Place;
}

module.exports.getRouteForAddress = async(destinationAddress, startAddressPlace, placeIndex, routeCalculator) => {
    const destinationPlace = await this.getPlaceFromAddress(destinationAddress, placeIndex);
    
    const params = {
        CalculatorName: routeCalculator,
        DeparturePosition: startAddressPlace.Geometry.Point,
        DestinationPosition: destinationPlace.Geometry.Point
    }

    const response = await locationClient.send(new CalculateRouteCommand(params));
    console.log(response);
    return response;
}
